compiler/damlc/daml-doc/test/DA/Daml/Doc/Render/Tests.hs

```diff
@@ -311,7 +372,71 @@ renderTest format externalAnchors (name, input) expected =
                  Html -> error "HTML testing not supported (use Markdown)"
     output = T.strip $ renderer input
     expect = T.strip expected
+  compareRendered output expect
+
+renderFolderTest ::
+     RenderFormat
+  -> AnchorMap
+  -> (String, [ModuleDoc])
+  -> (T.Text, Map Modulename T.Text)
+  -> Tasty.TestTree
+renderFolderTest format externalAnchors (name, input) expected =
+  testCaseSteps name $ \step -> do
+    let
+      modStep modName =
+        step ("Checking module '" <> T.unpack (unModulename modName) <> "'")
+
+      unexpectedMod modName output = do
+        modStep modName
+        T.putStrLn $ T.unlines
+          [ "Unexpected module in output:"
+          , "Expected: <nothing>"
+          , "Actual:"
+          , T.pack $ show output ]
+        assertFailure "Unexpected module in output."
+
+      missingMod modName expect = do
+        modStep modName
+        T.putStrLn $ T.unlines
+          [ "Expected module missing from output:"
+          , "Expected:"
+          , T.pack $ show expect
+          , "Actual: <nothing>" ]
+        assertFailure "Expected module missing from output."
+
```

Comment:
I think we should add a successful renderFolder output here as well.

Reply:
That would be `expectRstFolder` and `expectMarkdownFolder` from [here](https://github.com/digital-asset/daml/pull/14867/files#diff-3340a20963977e7e16c8eb8c0dcff9720f2a1d95cd11839dda12605fbda22469R36-R37). Do you mean they should be at the top level?

Reply:
I meant that I would expect to have a golden for the renderFolder/expectRst True output, unless I am missing something and there is something there.

Reply:
As discussed offline, we've agreed to keep the current test structure.